model/map/src/main/java/org/keycloak/models/map/lock/MapLockEntity.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.map.lock;
+
+import org.keycloak.models.map.annotations.GenerateEntityImplementations;
+import org.keycloak.models.map.common.AbstractEntity;
+import org.keycloak.models.map.common.DeepCloner;
+import org.keycloak.models.map.common.UpdatableEntity;
+import org.keycloak.storage.SearchableModelField;
+
+import java.util.Objects;
+
+/**
+ * Entity to hold locks needed for the {@link MapGlobalLockProvider}.
+ *
+ * @author Alexander Schwartz
+ */
+@GenerateEntityImplementations(
+        inherits = "org.keycloak.models.map.lock.MapLockEntity.AbstractLockEntity"
+)
+@DeepCloner.Root
+public interface MapLockEntity extends UpdatableEntity, AbstractEntity {
+
+    public static class SearchableFields {
+        public static final SearchableModelField<MapLockEntity> ID = new SearchableModelField<>("id", String.class);
+        public static final SearchableModelField<MapLockEntity> NAME = new SearchableModelField<>("name", String.class);
+    }
+
+    public abstract class AbstractLockEntity extends Impl implements MapLockEntity {
+
+        private String id;
+
+        @Override
+        public String getId() {
+            return this.id;
+        }
+
+        @Override
+        public void setId(String id) {
+            if (this.id != null) throw new IllegalStateException("Id cannot be changed");
+            this.id = id;
+            this.updated |= id != null;
+        }
+
+    }
+
+    String getName();
```

Comment:
@ahus1 Having `name` and `id` for this entity seems redundant as there are stores that don't need UUID as `id`. Can we have only `id` here and generate UUID only for JPA? This way we could also get rid of one `SearchableField`.

Reply:
The name is the name of the lock, which is defined by the caller. It can't be passed down as the ID to the store implementation, as the `#MapKeycloakTransaction#create()` is documented as follows: 

>  The ID of the entity passed in the parameter might change to a different value in the returned value if the underlying storage decided this was necessary.

As the name of the lock must be stored as-is in the store to effectively create a lock, the separate name field is required.

Reply:
You are right, I missed it. While I don't like the fact we have two searchable fields, where one of them will be probably never used, it seems there is not any way to use only `id` or only `name`. The only thing we can do is merge those on the storage layer and return the same for both `id` and `name`.